From 4ac81f8d9e0fa80ca255aab814cf971c61036d16 Mon Sep 17 00:00:00 2001
From: Charles Stevens <charles.stevens@logicpd.com>
Date: Thu, 10 Oct 2019 13:05:29 -0500
Subject: [PATCH 2/2] Enable FLL for sound card

---
 sound/soc/codecs/wm8962.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sound/soc/codecs/wm8962.c b/sound/soc/codecs/wm8962.c
index efd8910b1ff7..98e892a48fc4 100644
--- a/sound/soc/codecs/wm8962.c
+++ b/sound/soc/codecs/wm8962.c
@@ -3749,6 +3749,11 @@ static int wm8962_i2c_probe(struct i2c_client *i2c,
 		}
 	}
 
+    /* CHS : enable FLL */
+    regmap_update_bits(wm8962->regmap,
+                      WM8962_FLL_CONTROL_1,
+                      WM8962_FLL_ENA, WM8962_FLL_ENA);
+
 	pm_runtime_enable(&i2c->dev);
 	pm_request_idle(&i2c->dev);
 
-- 
2.17.1

