From 24102d59055c321553e7ef778af5dd722583d14b Mon Sep 17 00:00:00 2001
From: Charles Stevens <cstevens@beaconembedded.com>
Date: Mon, 24 Jan 2022 15:48:14 -0600
Subject: [PATCH] Implement GPIO poweroff for Beacon boards

Signed-off-by: Charles Stevens <cstevens@beaconembedded.com>
---
 plat/renesas/common/plat_pm.c | 31 +++++++------------------------
 1 file changed, 7 insertions(+), 24 deletions(-)

diff --git a/plat/renesas/common/plat_pm.c b/plat/renesas/common/plat_pm.c
index 6a9ad450d..6c8a2c756 100644
--- a/plat/renesas/common/plat_pm.c
+++ b/plat/renesas/common/plat_pm.c
@@ -162,30 +162,13 @@ finish:
 
 static void __dead2 rcar_system_off(void)
 {
-#if PMIC_ROHM_BD9571
-#if PMIC_LEVEL_MODE
-	if (rcar_iic_dvfs_send(PMIC, DVFS_SET_VID, DVFS_SET_VID_0V))
-		ERROR("BL3-1:Failed the SYSTEM-OFF.\n");
-#else
-	if (rcar_iic_dvfs_send(PMIC, BKUP_MODE_CNT, P_ALL_OFF))
-		ERROR("BL3-1:Failed the SYSTEM-RESET.\n");
-#endif
-#else
-	uint64_t cpu = read_mpidr_el1() & 0x0000ffff;
-	int32_t rtn_on;
-
-	rtn_on = rcar_pwrc_cpu_on_check(cpu);
-
-	if (cpu == rcar_boot_mpidr)
-		panic();
-
-	if (rtn_on)
-		panic();
-
-	rcar_pwrc_cpuoff(cpu);
-	rcar_pwrc_clusteroff(cpu);
-
-#endif /* PMIC_ROHM_BD9571 */
+	/* I stole these lines from pfc initialization code */
+#define GPIO_BASE               (0xE6050000U)
+#define GPIO_INOUTSEL6          (GPIO_BASE + 0x5404U)
+#define GPIO_OUTDT6             (GPIO_BASE + 0x5408U)
+	/* Assume power kill is output GP 6_11 (configured by BL2) */
+	/* Set GP 6_11 Low to power off */
+	mmio_write_32(GPIO_OUTDT6, (mmio_read_32(GPIO_OUTDT6)&~(1<<11)));
 	wfi();
 	ERROR("RCAR System Off: operation not handled.\n");
 	panic();
-- 
2.17.1

